lib/utils/responseHandler.js

```diff
@@ -9,7 +9,7 @@ module.exports = (response, body) => {
 
   if (response.statusCode != 200) {
     var errorResponse = outResponse;
-    if (response.headers['content-type'] === 'application/json;charset=UTF-8') {
+    if (/\bjson\b/.test(response.headers['content-type'])) {
```

Comment:
This was the reason we had to do the second parse of the message down below. We were checking for a very specific content-type which the wiremock diff does not supply.

Reply:
Since this is part of the SDK and the response may need to be both JSON and UTF-8 format, we should check with Steve before we make this change. The existence of a content-type header does not mean it is set to the correct type that can be parsed successful. At minimum, we may need to check that the content-type header has JSON as one of the items on the list of types.

Reply:
Okay, I'm not sure I completely understand what your suggested change is. Would you prefer we do something like this (in psedocode):

```javascript
if (JSON.isValidJSON(response.headers['content-type'])) { do thing }
```
I copied the logic used in Ruby so if we decide to change it we should update Ruby too.

I can ask Steve about the specific content-type when he comes in, but I'm sure the UTF-8 bit is not relevant. In node, we decode the body as UTF-8 by default for all but a couple of endpoints.

Reply:
I guess I am not sure that the syntax of this check does what I want.
It looks like it is checking that the response.header['content-type'] is a valid JSON syntax.
We need it to check that the value of the 'content-type' contains application/json.

I know what you put in the comment checks that it is JSON format. I am not sure if what is currently in the line of code currently is a string contains check or a object validation.

Reply:
Oh! Okay, the current line checks if the Content-Type header matches the regex /\bjson\b/. I copied the regex from Ruby. As far as I can tell, it checks if the Content-Type header contains the word 'json'. The \b's are to ensure json is not a substring of something

Reply:
I can modify it to see if the Content-Type contains 'application/json' if you want. That might be safer

Reply:
Since it is checking the string. application/json would be better.
http://www.iana.org/assignments/media-types/media-types.xhtml#application